src/storage/mod.rs

```diff
@@ -1516,6 +1516,14 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                         return Err(box_err!("Invalid KeyRanges"));
                     };
                     let mut result = Vec::new();
+                    let mut key_ranges = vec![];
+                    for range in &ranges {
+                        key_ranges.push(build_key_range(
+                            &range.start_key,
+                            &range.end_key,
+                            reverse_scan,
+                        ));
+                    }
```

Comment:
Why moving this part of the code to here?

Reply:
`take_start_key` in Line1529 will replace it with empty.

Reply:
I see, thx!